addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200GatewayHandler.java

```diff
@@ -0,0 +1,664 @@
+                        object.setUpdated(false);
+                        state = comm.getProvidersState(checkParameterReplacement(channel, device),
+                                channel.getAcceptedItemType(), getChannelConfigurationStrings(channel));
+                        if (state != null) {
+                            gatewayHandler.updateState(channel.getUID(), state);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * The sendKM200Thread class sends the data to the device.
+     *
+     * @author Markus Eckhardt
+     *
+     * @since 2.1.0
+     */
+    private static class SendKM200Thread extends Thread {
+
+        public SendKM200Thread(Map<Channel, byte[]> sendMap, KM200GatewayHandler gatewayHandler, KM200Device device,
+                KM200Comm<KM200Device> comm) {
+            super();
+            this.sendMap = sendMap;
+            this.gatewayHandler = gatewayHandler;
+            this.device = device;
+            this.comm = comm;
+        }
+
+        private Map<Channel, byte[]> sendMap = null;
+        private KM200GatewayHandler gatewayHandler;
+        private KM200Device device;
+        private KM200Comm<KM200Device> comm;
+
+        @Override
+        public void run() {
+            try {
+                logger.debug("Send-Thread started");
+                while (!isInterrupted()) {
+                    Map.Entry<Channel, byte[]> nextEntry = null;
+                    {
+                        /* Check whether a new entry is availible, if yes then take and remove it */
+                        synchronized (sendMap) {
+                            Iterator<Entry<Channel, byte[]>> i = sendMap.entrySet().iterator();
+
+                            if (i.hasNext()) {
+                                logger.debug("Send-Thread, new entry");
+                                nextEntry = i.next();
+                                i.remove();
+                            }
+                        }
+                    }
+
+                    if (nextEntry != null) {
+                        /* Now send the data to the device */
+                        Integer rCode;
+                        org.eclipse.smarthome.core.types.State state = null;
+                        Channel channel = nextEntry.getKey();
+                        byte[] encData = nextEntry.getValue();
+
+                        String service = checkParameterReplacement(channel, device);
+                        KM200CommObject object = device.getServiceObject(service);
+
+                        logger.debug("Sending: {}", service);
+
+                        if (object.getVirtual() == 1) {
+                            rCode = comm.sendDataToService(object.getParent(), encData);
+                        } else {
+                            rCode = comm.sendDataToService(service, encData);
+                        }
+                        logger.debug("Returncode: {}", rCode);
```

Comment:
the rCode is only used here for debugging. This could also be handled in the `comm` instance. this would clean up this method a little.